### PR TITLE
HDDS-8971. Example integration with Iceberg, Spark and Trino

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Create add-on for `ozone` docker-compose environment to demonstrate integration with [Iceberg](https://iceberg.apache.org/) and [Trino](https://trino.io/).

https://issues.apache.org/jira/browse/HDDS-8971

## How was this patch tested?

Added test script to verify the setup:
 * create S3 bucket in Ozone
 * create table and insert data in `spark-shell` (example taken from [Spark and Iceberg Quickstart](https://iceberg.apache.org/spark-quickstart/))
 * describe table and insert more data in `trino`
 * check that data/metadata is stored in Ozone

CI:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/5437385105
(Interesting part begins [here](https://github.com/adoroszlai/hadoop-ozone/actions/runs/5437385105/jobs/9888115026#step:5:102).)

### CODE REVIEW

**Review by ayushtkn — COMMENTED** [review]
Thanx @adoroszlai , just started exploring this.
A quick question: we are using ``tabulario`` image which isn't official but from a vendor, which we won't have any control, are we ok using that? 

Spark/Iceberg might be having official image may be, can we use that?

Hive does have official docker image, in case you want to explore: https://hub.docker.com/r/apache/hive

I tried iceberg with that
```
export HIVE_VERSION=4.0.0-alpha-2

docker run -d -p 10000:10000 -p 10002:10002 --env SERVICE_NAME=hiveserver2 --name hive4 apache/hive:${HIVE_VERSION}

 docker exec -it hive4 beeline -u 'jdbc:hive2://localhost:10000/'

 create table ice01 (id int) stored by iceberg;

show create table ice01;

insert into ice01 values (1),(2),(3),(4);

select * from ice01;
```

The ``show create table ice01;`` shows iceberg, which confirms the table is iceberg, I think I didn't see that it is mentioned anywhere, may be those guys configured some default or so. 

Show create output:
image

select query:
image

I think you are good with v1 table which doesn't support deletes/updates as in the current example in this PR.  (https://iceberg.apache.org/spec/#format-versioning)

It is pretty easy as well, just a tbl property and we are sorted for v2
```
create table ice02 (id int) stored by iceberg tblproperties ('format-version'='2');
```
so, we can do it in future as well. :-)

**Comment by adoroszlai** [comment]
Thanks @ayushtkn for starting to review.

> A quick question: we are using tabulario image which isn't official but from a vendor, which we won't have any control, are we ok using that?
> Spark/Iceberg might be having official image may be, can we use that?

I found this image from Tabular at https://iceberg.apache.org/spark-quickstart/ - if there was an official Apache Iceberg image, I guess they would have used that in the example.  I'm open to using any other image.  BTW, this is just a small experiment to help answer #4973.

Spark does have official images, will explore those.

**Comment by adoroszlai** [comment]
@SaketaChalamchala please take a look, too

**Comment by jojochuang** [comment]
There's a code against again the latest.

File: hadoop-ozone/dist/src/main/compose/ozone/test-iceberg.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    .writeTo("demo.nyc.taxis").append()
+
+spark.table("demo.nyc.taxis").show()
+EOF
+
+execute_command_in_container trino trino <<EOF
+DESCRIBE iceberg.nyc.taxis;
+INSERT INTO iceberg.nyc.taxis VALUES (2, 1000375, 7.2, 555, 'N');
+SELECT * FROM iceberg.nyc.taxis;
+EOF
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-ozone/dist/src/main/compose/ozone/test-iceberg.sh:65]: Thanks @SaketaChalamchala for the review.

Let's call it an Iceberg, Spark, Trino example instead. :)  (I followed the "Spark and Iceberg Quickstart" guide for the Iceberg part.)

**Comment by adoroszlai** [comment]
> There's a code against again the latest.

@jojochuang thanks for taking a look.  Conflict has been resolved.

**Review by jojochuang — APPROVED** [review]

**Review by jojochuang — CHANGES_REQUESTED** [review]
The PR looks good.

But I'm afraid of patent or license issues like hell. Looking at the source code for the tabulario/spark-iceberg docker image (https://github.com/tabular-io/docker-spark-iceberg/blob/main/docker-compose.yml) 

It includes MinIO and MinIO is AGPL. I want to make sure this is okay.

**Comment by adoroszlai** [comment]
@jojochuang we don't distribute MinIO in any way.  Users running this example download the MinIO docker image from Docker Hub.

But I'm fine abandoning this PR.

**Comment by jojochuang** [comment]
Hey @adoroszlai maybe it's time to revive it given all the interest around Trino? I'll just create an epic jira and put all Trino related work together.

**Comment by hli00** [comment]
For Ozone S3, another integration scenario is Trino-Hive-Ozone S3.

**Pull request closed without merging**